Updated How to run ACCESS-ESM with new repo

atteggiani opened the pull request:
- [x] Updated how to run ACCESS-ESM, using the updated GitHub Repo

heidinett commented on docs/models/run-a-model/run-access-esm.md:
```diff
@@ -32,32 +32,80 @@ module load conda/analysis3</code></pre>
+   Some modules may interfere with <code>git</code> commands (e.g., matlab/R2018a). If you have trouble cloning the repository, run the following command before trying again: <pre><code>module purge</code></pre>
+   After this step, don't forget to reload the <code>conda/analysis3</code> module to retrieve <code>payu</code>, as specified in the <a href="#model-specific-prerequisites">Model-specific prerequisites</a> section.
 </div>
+Different {{ model }} configurations are stored in different branches of {{ model }} GitHub repo.
```
add 'the' in the sentence, i.e.: "... stored in different branches of the {{ model }} GitHub repo."

heidinett commented on docs/models/run-a-model/run-access-esm.md:
```diff
@@ -32,32 +32,80 @@ module load conda/analysis3</code></pre>
+   After this step, don't forget to reload the <code>conda/analysis3</code> module to retrieve <code>payu</code>, as specified in the <a href="#model-specific-prerequisites">Model-specific prerequisites</a> section.
 </div>
+Different {{ model }} configurations are stored in different branches of {{ model }} GitHub repo.
+To check all the available branches on the repo, run the following command inside the newly created <code>access-esm</code> folder:
```
Edit as folllows: "To check all available branches in the repo, run.."

heidinett commented on docs/models/run-a-model/run-access-esm.md:
```diff
@@ -32,32 +32,80 @@ module load conda/analysis3</code></pre>
+The red-coloured branches are the available remote branches, formatted as <code>remotes/origin/&lt;branch-name&gt;</code>.
+To switch to a specific branch you can run the following command: 
+<pre><code>git checkout &lt;branch-name&gt;</code></pre>
+For example, the pre-industrial configuration of {{ model }} is available in the <code>pre-industrial</code> branch. To use the pre-industrial configuration we can run:
```
Edit: "To use the pre-industrial configuration, run:"

heidinett commented on docs/models/run-a-model/run-access-esm.md:
```diff
@@ -32,32 +32,80 @@ module load conda/analysis3</code></pre>
 ----------------------------------------------------------------------------------------
 
 ## Edit {{ model }} configuration
 It is good practice to create a new <i>git branch</i> to store all your modifications for a particular run, so as not to modify the reference configuration. 
```
For consistency, remove italics for "git branch"

heidinett commented on docs/models/run-a-model/run-access-esm.md:
```diff
@@ -32,32 +32,80 @@ module load conda/analysis3</code></pre>
 
-To create a local branch called <i>"example_run"</i>, from within the cloned repo execute:
-<pre><code>git checkout -b example_run</code></pre>
+To create a new branch called <i>"example_run"</i>, as a copy of the <code>pre-industrial</code> branch, from within the <code>access-esm</code> directory execute:
```
Edit: "To create a new branch called <i>"example_run"</i> as a copy of the <code>pre-industrial</code> branch, execute the following command from within the <code>access-esm</code> directory:"

heidinett commented on docs/models/run-a-model/run-access-esm.md:
```diff
@@ -356,12 +404,16 @@ For a complete documentation on how to use this framework, check the <a href="ht
+While the configuration is running, output files (and restart files) are moved from the <code>work</code> directory into the <code>archive</code> directory under <code>/scratch/$PROJECT/$USER/access-esm/archive/access-esm</code>. They are also symlinked in the <i>control</i> directory to <code>~/access-esm/archive</code>.
 <br>
-Both outputs and restarts are stored in subfolders for each different configuration (in this case, <code>esm-pre-industrial</code>). Inside the configuration folder, they are further subdivided for each internal run.
+Here, they are further subdivided for each internal run.
```
I would combine the above 2 sentences to read: "They are also symlinked in the <i>control</i> directory to <code>~/access-esm/archive</code>, where they are further subdivided for each internal run."

heidinett commented on docs/models/run-a-model/run-access-esm.md:
```diff
@@ -356,12 +404,16 @@ For a complete documentation on how to use this framework, check the <a href="ht
+<div class="note">
+    A run with a different {{ model }} configuration (different git branch) counts as a new internal run. 
+    <br>
+    Therefore, if output folders were already present, the new output folder will have its internal number set to the first available <i>XXX</i> number.
```
Edit: "Thus, if output folders already exist, the internal number of the new output folder will be set to the first available <i>XXX</i> number."

heidinett commented on docs/models/run-a-model/run-access-cm.md:
```diff
@@ -609,7 +609,7 @@ To investigate the cause of a failure, we need to look at the logs <code>job.err
 ### Model Live Diagnostics
 
-ACCESS-NRI developed the [Model Live Diagnostics](/model_evaluation/model_diagnostics) framework to check, monitor, visualise, and evaluate model behaviour and progress of ACCESS models currently running on <i>Gadi</i>.
+ACCESS-NRI developed the <a href="/model_evaluation/model_diagnostics">Model Live Diagnostics</a> framework to check, monitor, visualise, and evaluate model behaviour and progress of ACCESS models currently running on <i>Gadi</i>.
```
Edit: "ACCESS-NRI developed the <a href="/model_evaluation/model_diagnostics">Model Live Diagnostics</a> framework to check, monitor, visualise and evaluate model behaviour as well as the progress of ACCESS models running on <i>Gadi</i>."

heidinett reviewed (commented):
Only a few minor edits to the text required. See comments in relevant sections.

atteggiani commented:
Thank you for the review @heidinett!

ccarouge commented on docs/models/run-a-model/run-access-esm.md:
---
The payu stuff is not explained at all until way down the page and new comers will have difficulties to understand why it is in the prerequisites. I suggest adding just under the prerequisites title:

```
The {{ model }} is installed on the NCI supercomputer and uses <a href="https://payu.readthedocs.io/en/latest/" target="_blank"><i>payu</i></a>, a workflow management tool for running numerical models in supercomputing environments, to manage the model experiments. Following the prerequisites ensures you have access to this infrastructure.
```

---
I don't like where this note is placed:

        <div class="note">
            To run {{ model }}, you need to be a member of a project with allocated <i>Service Units</i> (SU). For more information, check <a href="/getting_started/first_steps#join-relevant-nci-projects">how to join relevant NCI projects</a>.
        </div>  

This should be moved to the prerequisites section, below the line:
```
Before running ACCESS-ESM, you need to fulfil general prerequisites outlined in the [First Steps](https://access-hive.org.au/getting_started/first_steps) section.
```
So people know what part of the First Steps is relevant.

---
At the end of the section on "Edit ACCESS-ESM configuration", I would add a subsection "Edit the model components configuration". In there I would put the text that is currently in the description of "submodels" with some small modifications:

```
Each model component contains additional configuration options that are read in when the model component is running, these options are typically useful to modify the physics used in the model or the input data. 
These options are specified in the subfolder of the control directory, whose name matches the submodel's name as specified in the <i>Master Configuration</i> (e.g., configuration options for the atmosphere submodel are in the ~/access-esm/esm-pre-industrial/atmosphere directory).
To modify these options please refer to the User Guide of each individual model components.
```

atteggiani replied:
> The payu stuff is not explained at all until way down the page and new comers will have difficulties to understand why it is in the prerequisites. I suggest adding just under the prerequisites title:
> 
> The {{ model }} is installed on the NCI supercomputer and uses <a href="https://payu.readthedocs.io/en/latest/" target="_blank"><i>payu</i></a>, a workflow management tool for running numerical models in supercomputing environments, to manage the model experiments. Following the prerequisites ensures you have access to this infrastructure.

Done.

> I don't like where this note is placed:
> 
>     <div class="note">
>         To run {{ model }}, you need to be a member of a project with allocated <i>Service Units</i> (SU). For more information, check <a href="/getting_started/first_steps#join-relevant-nci-projects">how to join relevant NCI projects</a>.
>     </div>  
> This should be moved to the prerequisites section, below the line:
> 
> Before running ACCESS-ESM, you need to fulfil general prerequisites outlined in the [First Steps](https://access-hive.org.au/getting_started/first_steps) section.
> So people know what part of the First Steps is relevant.

This is outlined in the First Steps, so users should already have done it when they get to this point. 
Anyway, in case they don't, there is the reference in the note with the link to the specific section [join relevant nci projects](https://access-hive.org.au/getting_started/first_steps#join-relevant-nci-projects)). 
I wouldn't specify this separately in the prerequisites.

> At the end of the section on "Edit ACCESS-ESM configuration", I would add a subsection "Edit the model components configuration". In there I would put the text that is currently in the description of "submodels" with some small modifications:
> 
> Each model component contains additional configuration options that are read in when the model component is running, these options are typically useful to modify the physics used in the model or the input data. 
> These options are specified in the subfolder of the control directory, whose name matches the submodel's name as specified in the <i>Master Configuration</i> (e.g., configuration options for the atmosphere submodel are in the ~/access-esm/esm-pre-industrial/atmosphere directory).
> To modify these options please refer to the User Guide of each individual model components.

Done.

ccarouge commented on docs/models/run-a-model/run-access-esm.md:
```diff
@@ -102,7 +150,7 @@ This file, which controls the general model configuration, contains several part
+walltime: 3:10:00</code></pre>
         These lines can be edited to change the <a href="https://opus.nci.org.au/display/Help/PBS+Directives+Explained" target="_blank">PBS directives</a> for the <a href="https://opus.nci.org.au/display/Help/4.+PBS+Jobs" target="_blank">PBS job</a>.
         <br>
         For example, to run {{ model }} under the <code>tm70</code> project (ACCESS-NRI), add the following line:
```
The note under this code section should say:
```
 The 'project' entry should always refer to a project with allocated <i>Service Units</i> if set explicitly. It will use your default project otherwise.
```

atteggiani replied:
The note currently says:

> To run ACCESS-ESM, you need to be a member of a project with allocated _Service Units_ (SU). For more information, check [how to join relevant NCI projects](https://access-hive.org.au/getting_started/first_steps#join-relevant-nci-projects).


Can we change your suggestion with the following?

> The `project` entry should always refer to a project with allocated _Service Units_ (SU), that you are a member of. If not set explicitly, ACCESS-ESM will run using your [default project](https://access-hive.org.au/getting_started/first_steps/#change-default-project-on-gadi) (this default project still needs to have allocated SU). For more information, check [how to join relevant NCI projects](https://access-hive.org.au/getting_started/first_steps#join-relevant-nci-projects).

ccarouge replied:
Yes, your proposed text is fine.

ccarouge commented on docs/models/run-a-model/run-access-esm.md:
```diff
@@ -32,32 +32,80 @@ module load conda/analysis3</code></pre>
+    <terminal-line>&emsp; pre-industrial</span></terminal-line>
+</terminal-window>
 
 ### Payu
 <a href="https://payu.readthedocs.io/en/latest/" target="_blank"><i>Payu</i></a> is a workflow management tool for running numerical models in supercomputing environments.
```
I find this section seems to be coming out of nowhere as is.  And since I suggest to put the definition of payu at the start, I propose to change to something like that:

```suggestion
### Payu data layout
Before modifying the {{ model }} configuration, you should understand how payu organises the inputs, configuration files, outputs and binary files.
```

atteggiani replied:
I think, despite the mention to payu at the beginning in the prerequisites, it is important to have a section that gives a bit more details, both on payu itself and on its data layout. 

In the section you suggested for the prerequisites I would add some minor specifics, something like:

> ACCESS-ESM is installed on NCI's supercomputer _Gadi_ and uses [payu](http://access-hive.org.au/models/run-a-model/run-access-esm/#payu), a tool for running and managing model experiments. Following these prerequisites ensures you have access to this infrastructure.

The link is not to payu website but to the more specific payu section below (the one you referenced in this comment), that instead links to payu's website and give more details about it.

ccarouge replied:
Yes, that's fine.

ccarouge commented on docs/models/run-a-model/run-access-esm.md:
```diff
@@ -356,12 +404,14 @@ For a complete documentation on how to use this framework, check the <a href="ht
-While the configuration is running, output files (and restart files) are moved from the <code>work</code> directory to the <code>archive</code> directory <code>/scratch/$PROJECT/$USER/access-esm/archive</code>. They are also symlinked in the <i>control</i> directory to <code>~/access-esm/esm-pre-industrial/archive</code>.
-<br>
-Both outputs and restarts are stored in subfolders for each different configuration (in this case, <code>esm-pre-industrial</code>). Inside the configuration folder, they are further subdivided for each internal run.
+While the configuration is running, output files (and restart files) are moved from the <code>work</code> directory into the <code>archive</code> directory under <code>/scratch/$PROJECT/$USER/access-esm/archive/access-esm</code>, where they are further subdivided for each internal run. They are also symlinked in the <i>control</i> directory to <code>~/access-esm/archive</code>
```
"While the configuration is running": is this really done during the run or at the end? Or is it during the run because it is done at the end of each internal run?

atteggiani replied:
I am actually not sure about this. Most likely is after each internal run.
In any case, might be safer to put "After the model run".

ccarouge replied:
Feel free to leave it as is if it's easier. As it is now, it feels like it is running in parallel to the model run which doesn't make much sense. But maybe it's simply because I'm too deeply technical on models...

atteggiani replied:
I changed it already, but I would not be too fussy about that anyway. 
I am not going to change it back for now.

ccarouge reviewed (changes requested):
I have commented beyond the modifications of this PR. So some of the comments might be hard to localise since I couldn't attach them to a code change. Happy to discuss if you need help understanding what I am referring to.

atteggiani commented on docs/models/run-a-model/run-access-esm.md:
```diff
@@ -32,32 +32,80 @@ module load conda/analysis3</code></pre>
-To create a local branch called <i>"example_run"</i>, from within the cloned repo execute:
-<pre><code>git checkout -b example_run</code></pre>
+To create a new branch called <i>"example_run"</i>, as a copy of the <code>pre-industrial</code> branch, from within the <code>access-esm</code> directory execute:
+<pre><code>git checkout -b example_run $(git log pre-industrial -1 --format=%H)</code></pre>
```
Makes sense!

atteggiani commented:
Thank you @ccarouge for the review.

I have addressed all the comments and commented on the ones I haven't or changed your suggestions. 

If you think it's ok to merge I'll merge it.
